### ISSUE TITLE
[2 beta 1] Error response from daemon: Ports are not available

### ISSUE THREAD

**Issue by Jan10** [craftcms/nitro#233]
### Description

I follow the [upgrading from Nitro 1](https://craftcms.com/docs/nitro/2.x/upgrade.html#back-up-databases-and-uninstall-nitro-1) guide. But after i run `nitro add` i get the following error:

```
 %  nitro add
Adding site…
Enter the hostname [craft-demo.nitro]: craft.test
  ✓ setting hostname to craft.test
  ✓ adding site ~/Sites/craft-demo
Enter the webroot for the site [web]:
  ✓ using webroot web
Choose a PHP version:
  1. 8.0
  2. 7.4
  3. 7.3
  4. 7.2
  5. 7.1
  6. 7.0
Enter your selection: 1
  ✓ setting PHP version 8.0
Error: there is no config file for the environment
```

When I setup nitro with `nitro init` I get the following error:

```
 %  nitro init
Setting up Nitro…
Would you like to use MySQL [Y/n] Y
Select the version of MySQL
  1. 8.0
  2. 5.7
  3. 5.6
Enter your selection: 1
Would you like to use PostgreSQL [Y/n] n
Would you like to use Redis [Y/n] n
Checking Nitro…
  ✓ network ready
  … pulling image ✓
  ✓ volume ready
  ✓ proxy ready
Checking network…
  ✓ network ready
Checking proxy…
  ✓ proxy ready
Checking databases…
  … checking mysql-8.0-3306.database.nitro ✗
Error: Error response from daemon: Ports are not available: listen tcp 127.0.0.1:3306: bind: address already in use
```

### Additional info

- Nitro CLI: 	 2.0.0-beta.1
- Nitro gRPC:
- Docker API: 	 1.41 (1.12 min)
- Docker CLI: 	 1.41

**Comment by Jan10** [comment]
@jasonmccallister thanks for the Feedback! I had mysql running via Brew.

**Comment by jasonmccallister** [comment]
Going to close this out @Jan10! Thank you for the report!

**Comment by olets** [comment]
@jasonmccallister I hit this too, likewise with Homebrew services. In my case the conflict was with redis:

```shell
% nitro init
# mariadb, no postgres, yes redis
Checking services…
  … checking dynamodb service ✓
  … checking mailhog service ✓
  … checking redis service Error: unable to start the container, Error response from daemon: Ports are not available: listen tcp 127.0.0.1:6379: bind: address already in use
% brew services ls
# -- snip --
redis         started
# -- snip --
% brew services stop redis
% nitro apply
# -- snip --
Checking services…
  … checking dynamodb service ✓
  … checking mailhog service ✓
  … checking redis service ✓
Checking proxy…
  … updating proxy ✓
Updating hosts file (you might be prompted for your password)
Adding sites to hosts file…
  … modifying hosts file ✓
Cleaning up...
Nitro is up and running 😃
```

To me this is a bug. My team currently uses an own-rolled docker-compose solution for Craft development. Our setup includes a redis service and a mysql service, and neither conflicts with host Homebrew services.

**Comment by jasonmccallister** [comment]
@olets to avoid confusion, we try to utilize default ports for services. This reduces the need for someone to make additional configuration changes to their code or GUI. We do have a way to [customize default ports](https://craftcms.com/docs/nitro/2.x/customizing.html#how-to-customize-nitro-s-defaults) for users who need to run something like redis outside of Nitro. If you need to keep a redis instance running outside of Nitro, those steps should help you set a custom port for the service.

If you are having an issue with ports, do you mind creating a new issue to track?
